openhcl/hcl/src/ioctl.rs

```diff
@@ -3346,4 +3367,31 @@ impl Hcl {
             hcl_kickcpus(self.mshv_vtl.file.as_raw_fd(), &data).expect("should always succeed");
         }
     }
+
+    /// Map or unmap guest device interrupt vector in VTL2 kernel
+    pub fn map_redirected_device_interrupt(
+        &self,
+        vector: u32,
+        apic_id: u32,
+        create_mapping: bool,
+    ) -> Option<u32> {
```

Comment:
In this case, it seems like the return value from the ioctl is important and we should return a `Result<u32, ioctl::Error>` right? we may want to know if in the teardown path fails, or whatever?

Reply:
The caller can choose to ignore the error and fallback anyways.